src/Orchard.Cms.Web/Modules/Orchard.Roles/Recipes/RolesStep.cs

```diff
@@ -0,0 +1,104 @@
+﻿using Orchard.Recipes.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Orchard.Recipes.Models;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Logging;
+using Microsoft.AspNetCore.Identity;
+using Orchard.Security;
+using Orchard.Security.Permissions;
+using Orchard.Environment.Extensions;
+
+namespace Orchard.Roles.Recipes
+{
+    public class RolesStep : RecipeExecutionStep
+    {
+        private readonly RoleManager<Role> _roleManager;
+        private readonly ITypeFeatureProvider _typeFeatureProvider;
+        private readonly IEnumerable<IPermissionProvider> _permissionProviders;
+
+
+        public RolesStep(RoleManager<Role> roleManager,
+                            ITypeFeatureProvider typeFeatureProvider,
+                            IEnumerable<IPermissionProvider> permissionProviders,
+                            ILoggerFactory loggerFactory, 
```

Comment:
The base should take a Logger, certainly not a LoggerFactory. Even though it's not your fault, please fix it.